From 2cd23391176a33f48b37becdce5b29bc61ffdee5 Mon Sep 17 00:00:00 2001
From: Biwen Li <biwen.li@nxp.com>
Date: Thu, 27 Jun 2019 15:02:44 +0800
Subject: [PATCH] arm64: dts: ls1012a/ls1043a/ls1046a/ls1088a/ls208xa: replace
 ftm0 with ftm_alarm0

commit b71736e1ffe7b7de89b1c85d88f2476b06cfc6ad from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The patch replaces ftm0 with ftm_alarm0 DT node
	- replace ftm0 with ftm_alarm0
	- add new rcpm node
	- remove old rcpm node
	- aliases ftm_alarm0 as rtc1

Signed-off-by: Biwen Li <biwen.li@nxp.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1012a.dtsi | 20 ++++++++--------
 .../dts/freescale/fsl-ls1043a-qds-sdk.dts     |  2 +-
 .../dts/freescale/fsl-ls1043a-rdb-sdk.dts     |  2 +-
 .../arm64/boot/dts/freescale/fsl-ls1043a.dtsi | 24 +++++++++++--------
 .../dts/freescale/fsl-ls1046a-frwy-sdk.dts    |  2 +-
 .../dts/freescale/fsl-ls1046a-qds-sdk.dts     |  2 +-
 .../dts/freescale/fsl-ls1046a-rdb-sdk.dts     |  2 +-
 .../arm64/boot/dts/freescale/fsl-ls1046a.dtsi | 23 +++++++++++-------
 .../arm64/boot/dts/freescale/fsl-ls1088a.dtsi | 22 ++++++++++-------
 .../arm64/boot/dts/freescale/fsl-ls208xa.dtsi | 15 ++++++++----
 10 files changed, 68 insertions(+), 46 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
index b3b4948a2f64..2288ac265c5a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
@@ -23,6 +23,7 @@
 		rtic-c = &rtic_c;
 		rtic-d = &rtic_d;
 		sec-mon = &sec_mon;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -289,11 +290,16 @@
 			#thermal-sensor-cells = <1>;
 		};
 
-		ftm0: ftm0@29d0000 {
+		rcpm: rcpm@1ee2140 {
+			compatible = "fsl,ls1012a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1ee2140 0x0 0x4>;
+			#fsl,rcpm-wakeup-cells = <1>;
+		};
+
+		ftm_alarm0: timer@29d0000 {
 			compatible = "fsl,ls1012a-ftm-alarm";
-			reg = <0x0 0x29d0000 0x0 0x10000>,
-			      <0x0 0x1ee2140 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
+			reg = <0x0 0x29d0000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x20000>;
 			interrupts = <0 86 0x4>;
 			big-endian;
 		};
@@ -497,12 +503,6 @@
 					<0000 0 0 4 &gic 0 113 IRQ_TYPE_LEVEL_HIGH>;
 			status = "disabled";
 		};
-
-                rcpm: rcpm@1ee2000 {
-                        compatible = "fsl,ls1012a-rcpm", "fsl,qoriq-rcpm-2.1";
-                        reg = <0x0 0x1ee2000 0x0 0x1000>;
-                        fsl,#rcpm-wakeup-cells = <1>;
-                };
        };
 
        reserved-memory {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
index e8254d95f4da..a8d26c7ff58f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
@@ -189,7 +189,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts
index 2e1b8d36e3b6..0ecb90f7e372 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts
@@ -222,7 +222,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index 117d545efe91..3366f316c8a2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -27,6 +27,7 @@
 		ethernet4 = &enet4;
 		ethernet5 = &enet5;
 		ethernet6 = &enet6;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -657,16 +658,6 @@
 			status = "disabled";
 		};
 
-		ftm0: ftm0@29d0000 {
-			compatible = "fsl,ls1043a-ftm-alarm";
-			reg = <0x0 0x29d0000 0x0 0x10000>,
-			      <0x0 0x1ee2140 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
-			interrupts = <0 86 0x4>;
-			big-endian;
-			status = "okay";
-		};
-
 		wdog0: wdog@2ad0000 {
 			compatible = "fsl,ls1043a-wdt", "fsl,imx21-wdt";
 			reg = <0x0 0x2ad0000 0x0 0x10000>;
@@ -875,6 +866,19 @@
 			status = "disabled";
 		};
 
+		rcpm: rcpm@1ee2140 {
+			compatible = "fsl,ls1043a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1ee2140 0x0 0x4>;
+			#fsl,rcpm-wakeup-cells = <1>;
+		};
+
+		ftm_alarm0: timer@29d0000 {
+			compatible = "fsl,ls1043a-ftm-alarm";
+			reg = <0x0 0x29d0000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x20000>;
+			interrupts = <0 86 0x4>;
+			big-endian;
+		};
 	};
 
 	firmware {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
index c46a14a091b6..8c0e8aaddc2f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
@@ -196,7 +196,7 @@
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
index b2d205d3ae38..0c2e0623ee78 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
@@ -186,7 +186,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
index 0fcdc50281ca..d38e57982cf0 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
@@ -229,7 +229,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
index 37c8e5e287c7..eb3169cf882c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
@@ -28,6 +28,7 @@
 		ethernet5 = &enet5;
 		ethernet6 = &enet6;
 		ethernet7 = &enet7;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -560,15 +561,6 @@
 			status = "disabled";
 		};
 
-		ftm0: ftm0@29d0000 {
-			compatible = "fsl,ls1046a-ftm-alarm";
-			reg = <0x0 0x29d0000 0x0 0x10000>,
-			      <0x0 0x1ee2140 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
-			interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
-			big-endian;
-		};
-
 		wdog0: watchdog@2ad0000 {
 			compatible = "fsl,imx21-wdt";
 			reg = <0x0 0x2ad0000 0x0 0x10000>;
@@ -814,6 +806,19 @@
 			status = "disabled";
 		};
 
+		rcpm: rcpm@1ee208c {
+			compatible = "fsl,ls1046a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1ee208c 0x0 0x4>;
+			#fsl,rcpm-wakeup-cells = <1>;
+		};
+
+		ftm_alarm0: timer@29d0000 {
+			compatible = "fsl,ls1046a-ftm-alarm";
+			reg = <0x0 0x29d0000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x20000>;
+			interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
+			big-endian;
+		};
 	};
 
 	reserved-memory {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
index b53671e7dfbb..ff1a3068835a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
@@ -18,6 +18,7 @@
 
 	aliases {
 		crypto = &crypto;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -325,14 +326,6 @@
 			status = "disabled";
 		};
 
-		ftm0: ftm0@2800000 {
-			compatible = "fsl,ls1088a-ftm-alarm";
-			reg = <0x0 0x2800000 0x0 0x10000>,
-			      <0x0 0x1e34050 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
-			interrupts = <0 44 4>;
-		};
-
 		i2c0: i2c@2000000 {
 			compatible = "fsl,vf610-i2c", "fsl,ls1088a-vf610-i2c";
 			#address-cells = <1>;
@@ -864,6 +857,19 @@
                                      <0 208 4>,
                                      <0 209 4>;
                 };
+
+		rcpm: rcpm@1e34040 {
+			compatible = "fsl,ls1088a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1e34040 0x0 0x18>;
+			#fsl,rcpm-wakeup-cells = <6>;
+		};
+
+		ftm_alarm0: timer@2800000 {
+			compatible = "fsl,ls1088a-ftm-alarm";
+			reg = <0x0 0x2800000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x0 0x0 0x0 0x0 0x4000 0x0>;
+			interrupts = <0 44 4>;
+		};	
 	};
 
 	firmware {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
index 17c2dc79b619..42d04a2e2b34 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
@@ -24,6 +24,7 @@
 		serial1 = &serial1;
 		serial2 = &serial2;
 		serial3 = &serial3;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpu: cpus {
@@ -752,11 +753,17 @@
 			interrupts = <0 12 4>;
 		};
 		
-		ftm0: ftm0@2800000 {
+		rcpm: rcpm@1e34040 {
+			compatible = "fsl,ls208xa-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1e34040 0x0 0x18>;
+			#fsl,rcpm-wakeup-cells = <6>;
+		};
+
+		ftm_alarm0: timer@2800000 {
 			compatible = "fsl,ls208xa-ftm-alarm";
-                        reg = <0x0 0x2800000 0x0 0x10000>,
-                              <0x0 0x1e34050 0x0 0x4>;
-                        reg-names = "ftm", "pmctrl";
+			reg-names = "ftm", "pmctrl";
+			reg = <0x0 0x2800000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x0 0x0 0x0 0x0 0x4000 0x0>;
 			interrupts = <0 44 4>;
 		};
 	};
-- 
2.17.1

